lib/Dialect/ESI/capnp/Schema.cpp

```diff
@@ -258,6 +271,161 @@ bool TypeSchemaImpl::operator==(const TypeSchemaImpl &that) const {
   return type == that.type;
 }
 
+//===----------------------------------------------------------------------===//
+// Capnp encode / decode RTL builders.
+//
+// These have the potential to get large and complex as we add more types. The
+// encoding spec is here: https://capnproto.org/encoding.html
+//===----------------------------------------------------------------------===//
+
+static size_t bits(::capnp::schema::Type::Reader type) {
+  using ty = ::capnp::schema::Type;
+  switch (type.which()) {
+  case ty::VOID:
+    return 0;
+  case ty::UINT8:
+  case ty::INT8:
+    return 8;
+  case ty::UINT16:
+  case ty::INT16:
+    return 16;
+  case ty::UINT32:
+  case ty::INT32:
+    return 32;
+  case ty::UINT64:
+  case ty::INT64:
+    return 64;
+  default:
+    assert(false && "Type not yet supported");
+  }
+}
+
+/// Build an RTL/SV dialect capnp encoder for this type.
```

Comment:
I believe this is encoding messages without any packing, should that be documented?

Reply:
What do you mean by "without any packing"? Do you mean in the capnp message? Do you mean on the RTL/SV side?

Reply:
I think you mean on the RTL side, yes? Concepts like padding and packing aren't generally a think in hardware -- unpacked structs and padded integers are simulation concepts, so I thought this was implied. Remember, this particular code isn't just for simulation! (There's a reason I'm doing this decoding in hardware.) I'll add some documentation anyway.

Reply:
Sorry, I was actually referring to the capnp concept of packing: https://capnproto.org/encoding.html#packing. Is it worth calling out anything about this (or any other specifics of that document)?